5-Cookies/client-react/src/components/TodoList/TodoListItem/TodoListItem.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState, FC } from "react";
+import { ITodoItem, TodoListItem } from "../../../interfaces/interfaces";
+import { classes } from "../../../js-styles/style";
+import IconButton from "../../general/IconButton";
+import Switch from "../../general/Switch";
+import EditTodoItem from "../EditTodoItem";
+import { todosApi } from "../../../../App";
+import dataHooks from "../../../dataHooks/dataHooks";
+import "../../../../style.css";
+import { Item } from "../../../interfaces/interfaces";
+import { Guid } from "../../../../../common/interfaces/Todo";
+
+const TodoListItem: FC<ITodoItem> = ({ item, removeTodo }) => {
+  const [isEditMode, setIsEditMode] = useState<boolean>(false);
+  const [todoItem, setTodoItem] = useState<Item>(item);
+
+  const dispatchEditTodo = async (newTodo: Item, id: Guid): Promise<void> => {
+    try {
+      await todosApi.editTodo(newTodo, id);
+    } catch (error) {
+      console.log(error);
```

Comment:
why swallowing the error?

Reply:
Actually, I just changed that to toast an error with `react-toastify` library to alert the user of a problem.